### PR TITLE
Added Tests that use Halibut in a very similar way to Server to Tentacle communication

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

Added Tests that use Halibut in a very similar way to Server to Tentacle communication

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+{
+    public class WhenCallingServicesSimilarToTheOnesInTentacle : BaseTest
+    {
+        [Test]
+        [LatestAndPreviousClientAndServiceVersionsTestCases]
+        public async Task FilesCanBeDownloaded(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .WithTentacleServices()
+                       .WithHalibutLoggingLevel(LogLevel.Trace)
```
Comment (LukeButters) [inline source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:29]: trace?

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            }
+        }
+
+        [Test]
+        [LatestAndPreviousClientAndServiceVersionsTestCases]
+        public async Task FilesCanBeUploaded(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .WithTentacleServices()
+                       .WithHalibutLoggingLevel(LogLevel.Trace)
```
Comment (LukeButters) [inline source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:63]: trace?

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            }
+        }
+
+        [Test]
+        [LatestAndPreviousClientAndServiceVersionsTestCases]
+        public async Task ScriptCanBeExecutedWithScriptServiceV1(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .WithTentacleServices()
+                       .WithHalibutLoggingLevel(LogLevel.Trace)
```
Comment (LukeButters) [inline source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:106]: trace?

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+            }
+        }
+
+        [Test]
+        [LatestAndPreviousClientAndServiceVersionsTestCases]
+        public async Task ScriptCanBeExecutedWithScriptServiceV2(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .WithTentacleServices()
+                       .WithHalibutLoggingLevel(LogLevel.Trace)
```
Comment (LukeButters) [inline source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:152]: trace?

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        public async Task FilesCanBeDownloaded(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .WithTentacleServices()
+                       .WithHalibutLoggingLevel(LogLevel.Trace)
+                       .Build(CancellationToken))
+            {
+                var fileTransferService = clientAndService.CreateClient<IFileTransferService>();
+
+                DownloadFile(fileTransferService);
```
Comment (LukeButters) [inline source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:34]: Interesting that this is the same number lines if we used a for loop.

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+                var capabilitiesService = clientAndService.CreateClient<ICapabilitiesServiceV2>();
+
+                var scriptBody = GetRandomMultiLineString();
+                var startScriptCommand = new StartScriptCommand(scriptBody, ScriptIsolationLevel.NoIsolation, TimeSpan.MaxValue, null, null, taskId: Guid.NewGuid().ToString());
+
+                try
+                {
+                    capabilitiesService.GetCapabilities();
+                }
+                catch { }
```
Comment (LukeButters) [inline source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:119]: I wonder why this is caught

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+
+                var complete = false;
+                var logs = new List<ProcessOutput>();
+                long nextLogSequence = 0;
+
+                try
+                {
+                    capabilitiesService.GetCapabilities();
+                }
+                catch {}
```
Comment (LukeButters) [inline source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:170]: Same here with catching this call....

**Review by LukeButters — APPROVED** [review]

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs (hunk 7 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:170]: Not needed, brain not working

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs (hunk 6 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:119]: Not needed, brain not working

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs (hunk 4 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:152]: Fixed

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs (hunk 3 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:106]: Fixed

File: source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs (hunk 2 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Tentacle/WhenCallingServicesSimilarToTheOnesInTentacle.cs:63]: Fixed

File: source/Halibut.TestUtils.Contracts/Tentacle/Contracts/TentacleContracts.cs [hunk 8]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace Octopus.Tentacle.Contracts
+{
+    internal static class TentacleContracts
```
[the remark this answers is not in the document]
Comment (nathanwoctopusdeploy) [reply source/Halibut.TestUtils.Contracts/Tentacle/Contracts/TentacleContracts.cs:5]: No, good spot

**Comment by nathanwoctopusdeploy** [comment]
[sc-53159]

**Pull request merged**